redpanda/utils/utils.go

```diff
@@ -176,9 +176,22 @@ func ConnectionTypeToString(t controlplanev1beta2.Cluster_ConnectionType) string
 // TypeListToStringSlice converts a types.List to a []string, stripping
 // surrounding quotes for each element.
 func TypeListToStringSlice(t types.List) []string {
-	var s []string
+	if t.IsNull() {
+		return nil
+	}
+	s := []string{}
 	for _, v := range t.Elements() {
-		s = append(s, strings.Trim(v.String(), "\"")) // it's easier to strip the quotes than type converting until you hit something that doesn't include them
+		stringval, ok := v.(types.String)
+		if ok {
+			s = append(s, stringval.ValueString())
+		} else {
+			// TODO: how can we make this more type-safe? ideally this function is
+			// only ever called on types.List that actually hold strings. The Cloudflare
+			// Terraform provider has an interesting solution that defines generics that
+			// wrap types.List but also hold type information, and can define strongly-typed
+			// functions on those.
```

Comment:
This should be an issue not a TODO

Reply:
How do you want that organized? Just an issue? Issue + a single TODO line that links to the issue?

Reply:
That seems like a reasonable way to handle it